Jenkinsfile

```diff
@@ -5,61 +5,73 @@
 // sh invocations in pipeline scripts.
 // See https://issues.jenkins-ci.org/browse/JENKINS-37116
 
-node {
-  try {
+pipeline {
+    agent any
+    stages {
 
-    slackNotify("STARTED")
+        stage('Prepare build') {
+            steps {
+                slackNotify("STARTED")
+                checkout scm
+                sh "spynl/cli/ops/prepare-stage.sh -u $scm_urls -r $revision -f $fallbackrevision -m"
+            }
+            post {
+                always {
+                    archive 'repo-state.txt'
+                }
+            }
+        }
 
-    // Stamp exact states of repos and run unit tests
-    stage('Unit Tests') {
-      sh 'rm -rf repo-state.txt venv'
-      checkout scm
-      sh """
-          spynl/cli/ops/prepare-stage.sh -u $scm_urls -r $revision -f $fallbackrevision -m && \\
-          source venv/bin/activate && pip install coverage pylint pytest-cov && mkdir -p pylint-results && \\
-          spynl dev.test --reports
-      """
-      archive 'repo-state.txt'                              // for history
-      stash name:'repo-state', includes:'**/repo-state.txt' // for using it again later in this build
-      junit 'venv/**/pytests.xml'
-      //coverage 'venv/**/coverage.xml'  // Check this ticket: https://issues.jenkins-ci.org/browse/JENKINS-30700
-    }
+        stage('Unit Tests') {
+            steps {
+                sh "source venv/bin/activate && pip install pytest-cov && spynl dev.test --reports"
+            }
+            post {
+                always {
+                    junit 'venv/**/pytests.xml'
+                    //coverage 'venv/**/coverage.xml'  // Check this ticket: https://issues.jenkins-ci.org/browse/JENKINS-30700
```

Comment:
we publish converage raport in html on v6:
https://jenkins.swcloud.nl/job/softwear6/job/build/Unit_test_coverage/
it's done like this:
https://bitbucket.org/softwear/softwear6/commits/940be7e8ccb7cee593d7475c46f38432b194b788

Reply:
I will come back to this as I dont think the little graph works for spynl.
Ill try to fix it in the next iteration, thanks.